AdSpirit Bid Adapter : updated version with native support

carsten1980 opened the pull request:
<!--
Thank you for your pull request! 

Please title your pull request like this: 'Module: Change', eg 'Adspirit Bid Adapter: support for Native'

Please make sure this PR is scoped to one change or you may be asked to resubmit. 
 
Please make sure any added or changed code includes tests with greater than 80% code coverage. 

See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.

For any user facing change, submit a link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: also submit your bidder parameter documentation as noted in https://docs.prebid.org/dev-docs/bidder-adaptor.html#submitting-your-adapter -->
- [x] Updated bidder adapter  <!--  IMPORTANT: (1) consider whether you need to upgrade your bidder parameter documentation in https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders and (2) if you have a Prebid Server adapter, please consider whether that should be updated as well. --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
<!-- Describe the change proposed in this pull request -->
1. We added native support and updated the bidadapter source code with new test cases.

<!-- For new bidder adapters, please provide the following
- contact email of the adapter’s maintainer
- test parameters for validating bids:
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/hello_world.html) sample page. -->


## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->

ChrisHuie commented:
@carsten1980 it looks like our circleci tests are not running on this pr. Usually this is because you are following your fork in circleci. If you could unfollow your fork and follow the main prebid.js project that should kick off the tests. Tried to do it manually but couldnt

carsten1980 commented:
@ChrisHuie I have unfollowed my fork in CircleCI as suggested. Could you please check if the tests are now running? If not, let me know if there's anything else I need to do to trigger them. Thanks

patmmccann commented:
You must fix those errors to merge

carsten1980 commented:
@patmmccann @ChrisHuie I updated the JS and spec files and tested them locally without any errors. Could you please let me know if anything else is needed for PR approval?

ChrisHuie commented:
@carsten1980 still can't get our testing to run on it in here but will run them locally and review

carsten1980 commented:
Hi @ChrisHuie , we have removed the .circleci folder and updated the branch.

carsten1980 commented:
Hi @ChrisHuie Could you please hold off on merging this PR for this week? We are incorporating new OpenRTB standards into our script and will provide the updated code on the next working day.

ChrisHuie commented on .circleci/config.yml:
```diff
@@ -1,75 +0,0 @@
-# Javascript Node CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-javascript/ for more details
-#
```
Please undo changes to this file or remove from files tracked in pr please

carsten1980 commented:
Hi @ChrisHuie I have added .circleci/ to .gitignore and removed the .circleci folder, then updated the PR. Could you please check it and let me know if there's anything else I need to do for a successful merge?

carsten1980 commented:
Hello @ChrisHuie , I've updated the assets fields in the latest version. Could you please review and merge this PR? Let me know if any changes are needed from our side. Thanks!

carsten1980 commented:
Hi @ChrisHuie ,I’ve updated the JS file to resolve the lint errors, and I'm no longer seeing any issues.
Could you please review and merge this PR? Let me know if any changes are needed from our side.

carsten1980 commented:
Hello @ChrisHuie @patmmccann , could you please review our latest code and merge it as early as possible? I'm eagerly waiting for your response. Let me know if I need to make any changes.

patmmccann commented on .gitignore:
```diff
@@ -80,3 +80,5 @@ typings/
 
 # MacOS system files
 .DS_Store
+#circleci ignore
```
can you remove this?

carsten1980 replied:
Removed the #circleci ignore comment.

patmmccann commented on modules/adspiritBidAdapter.js:
```diff
@@ -9,7 +11,7 @@ export const spec = {
   code: 'adspirit',
   aliases: ['twiago'],
-  supportedMediaTypes: [BANNER, NATIVE],
+  supportedMediaTypes: [BANNER, NATIVE, VIDEO],
```
you're adding video but the title suggests you are adding native?

carsten1980 replied:
Currently, we are adding support for Native only. We have plans to add Video in the coming days, so we’have included it in the array in advance.

carsten1980 commented on .circleci/config.yml:
```diff
@@ -1,75 +0,0 @@
-# Javascript Node CircleCI 2.0 configuration file
```
Remove circleci.

carsten1980 replied:
Circle is giving some errors, so we have decided to remove it from the project.

carsten1980 replied:
I've now added the config.yml file back to the project.

carsten1980 commented:
Hi @patmmccann @ChrisHuie CircleCI did not run because the 'JavaScript Node CircleCI 2.0 configuration file' has been removed from the project due to some errors. What do you suggest we do? Is it possible to test this locally? Any suggestions would be helpful so we can complete this merge sooner.

carsten1980 commented:
Hi @patmmccann @ChrisHuie 

I’ve now added the config.yml file back to the project. All checks have passed successfully, and I don’t see any errors on my end. Could you please have a look from your side and let me know if I still need to make any changes before this PR can be merged?

Thanks

carsten1980 commented:
> circleci did not run

Please check it now

carsten1980 commented:
Hi @patmmccann @ChrisHuie ,
I've made all the requested changes and everything looks good with no errors on my end.
Please review and merge this PR at your earliest convenience. If any further changes are needed, I'm happy to make them.

Thank you!

ChrisHuie commented:
@carsten1980 can you undo changes to the . files 

Looks like you accidentally updated .circleci/ and .gitignore files

ChrisHuie commented:
@carsten1980 yeah. There is just no reason to update those files. If we merge currently there is a newly created .circleci/ file and an extra line added to the .gitignore. So just limits this pr to the 3 main adspirit files changed.

carsten1980 commented:
Hi @ChrisHuie ,
I’ve untracked the .circleci file from the project, so the PR now only includes the three updated adspirit files.Could you please take a look and let me know the current status of the PR?

Thanks!

carsten1980 commented:
I accidentally click on the merge close button. So I am reopening this merge request again.

carsten1980 commented:
I am reopening this merge request.

patmmccann commented on modules/adspiritBidAdapter.js:
```diff
@@ -9,7 +9,7 @@ export const spec = {
   code: 'adspirit',
   aliases: ['twiago'],
-  supportedMediaTypes: [BANNER, NATIVE],
+  supportedMediaTypes: [BANNER, NATIVE,VIDEO],
```
this is causing a linting error, add a space

carsten1980 commented:
Hi @ChrisHuie @patmmccann , 
I’ve added all three main files for the AdSpirit Bid Adapter and resolved all lint errors. Could you please review and merge this PR? 

Thank you
